src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -75,35 +76,62 @@ internal sealed class HttpWebRequestDiagnosticSource : DiagnosticListener
+            }
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
+            activity.DisplayName = HttpTagHelper.GetOperationNameForHttpMethod(request.Method);
+
+            activity.AddTag(SpanAttributeConstants.ComponentKey, "http");
+            activity.AddTag(SpanAttributeConstants.HttpMethodKey, request.Method);
+            activity.AddTag(SpanAttributeConstants.HttpHostKey, HttpTagHelper.GetHostTagValueFromRequestUri(request.RequestUri));
+            activity.AddTag(SpanAttributeConstants.HttpUrlKey, request.RequestUri.OriginalString);
+            activity.AddTag(SpanAttributeConstants.HttpFlavorKey, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.ProtocolVersion));
+
+            InstrumentRequest(request, activity);
+
+            activity.SetCustomProperty("HttpWebRequest.Request", request);
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddResponseTags(HttpWebResponse response, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
+            activity.AddTag(SpanAttributeConstants.HttpStatusCodeKey, HttpTagHelper.GetStatusCodeTagValueFromHttpStatusCode(response.StatusCode));
+            activity.AddTag(SpanAttributeConstants.HttpStatusTextKey, response.StatusDescription);
+
+            activity.SetCustomProperty("HttpWebRequest.Response", response);
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddExceptionTags(Exception exception, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer);
-            this.Initialize();
-            return result;
+            if (exception is WebException wexc && wexc.Response is HttpWebResponse response)
+            {
+                activity.AddTag(SpanAttributeConstants.HttpStatusCodeKey, HttpTagHelper.GetStatusCodeTagValueFromHttpStatusCode(response.StatusCode));
+                activity.AddTag(SpanAttributeConstants.HttpStatusTextKey, response.StatusDescription);
+            }
+            else
+            {
+                activity.AddTag("error", exception.Message);
```

Comment:
this is not a wellknown tagname right? Were we doing this previously? Most exporters export selected, well-known tags, so this may not be used

Reply:
This is due to the `Status` gap with Span. Previous logic is here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/acaec5d9ab454cc3252c37d2c266fb0fee517d65/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestDiagnosticListener.net461.cs#L123-L161

Today we get `exception.Message` as Status.Description ("ot.status_description") in cases where the exception wasn't tied to an Http status code. Agree "error" feels weird but not sure what to do. I could switch to "ot.status_description" but that also feels weird without "ot.status_code" also being sent? This was more or less a temporary thing while we wait to find out what we'll be doing about the Status gap.

Reply:
Agreed: this is temporary while OTel defines status and error reporting. That said `error` is a semantic tag in OpenTracing - https://docs.google.com/spreadsheets/d/1c8QDEBrN6ihVlTiz7Ak_CE-o2flR7XgdT8yFQeolPRc/edit#gid=2124992250 so it seems better to avoid it.

For the time being, setting both `ot.status_description` and `ot.status_code` seem reasonable to me.

Reply:
Agree about setting ot.status_description and ot.status_code

Reply:
Updated